src/stacks-hierarchy/saved-payment-utils.ts

```diff
@@ -32,8 +32,20 @@ export function usePreviousPaymentMethods(): {
         if (!recurringPayment) return false;
       }
 
-      // Payment type is not enabled
-      if (!paymentTypes.includes(paymentMethod.paymentType)) return false;
+      const enabledPaymentTypes: PaymentType[] = Array.from(
+        new Set(paymentTypes),
+      );
+      const paymentCardTypes: PaymentType[] = [
+        PaymentType.Amex,
+        PaymentType.Visa,
+        PaymentType.Mastercard,
+      ];
+
+      if (paymentTypes.some((type) => paymentCardTypes.includes(type))) {
+        enabledPaymentTypes.push(PaymentType.PaymentCard);
+      }
+      if (!enabledPaymentTypes.includes(paymentMethod.paymentType))
```

Comment:
Thinking about this once more: Could it make sense to make a util function for some of this? Just so it's easier to understand what's being done.

E.g. `hasPaymentCard(paymentTypes: PaymentType)`